exercises/sam_jenkins/ex02_oo_basics/src/main.cpp

```diff
@@ -19,16 +19,16 @@ int main(int, char **)
 	vector<unique_ptr<Shapes>> shapes;
 	shapes.reserve(10);
 
-	shapes.emplace_back(make_unique<Square>(5));
-	shapes.emplace_back(make_unique<Circle>(7));
-	shapes.emplace_back(make_unique<Rectangle>(10, 5));
-	shapes.emplace_back(make_unique<Triangle>(10, 5));
-	shapes.emplace_back(make_unique<Square>(9));
-	shapes.emplace_back(make_unique<Circle>(3));
-	shapes.emplace_back(make_unique<Rectangle>(2, 4));
-	shapes.emplace_back(make_unique<Triangle>(5, 6));
-	shapes.emplace_back(make_unique<Square>(1));
-	shapes.emplace_back(make_unique<Circle>(2));
+	shapes.push_back(unique_ptr<Square>(new Square(5)));
+	shapes.push_back(unique_ptr<Circle>(new Circle(7)));
+	shapes.push_back(unique_ptr<Rectangle>(new Rectangle(10, 5)));
+	shapes.push_back(unique_ptr<Triangle>(new Triangle(10, 5)));
+	shapes.push_back(unique_ptr<Square>(new Square(9)));
+	shapes.push_back(unique_ptr<Circle>(new Circle(3)));
+	shapes.push_back(unique_ptr<Rectangle>(new Rectangle(2, 4)));
+	shapes.push_back(unique_ptr<Triangle>(new Triangle(5, 6)));
+	shapes.push_back(unique_ptr<Square>(new Square( 1)));
+	shapes.push_back(unique_ptr<Circle>(new Circle(2)));
```

Comment:
Fair enough, annoyingly looking at the RHEL build you're on GCC 4.8.5 which doesn't understand c++14. [4.9.0 does however...](https://godbolt.org/g/2akZC4)

Reply:
Note that this is worse though as the `new` could potentially throw and leak memory. Using `make_unique` avoids that possibility.

Reply:
yeah I understand, but since make_unique isn't in c++ 11 I couldn't find another way to do it without implementing it myself